packages/office-ui-fabric-react/src/components/Panel/Panel.styles.ts

```diff
@@ -0,0 +1,342 @@
+//       height: 'auto',
+//       width: '44px',
+//       color: theme.palette.neutralSecondary,
+//       fontSize: IconFontSizes.large
+//     },
+//     rootHovered: {
+//       color: theme.palette.neutralPrimary
+//     }
+//   });
+// }
+
+export const getStyles = (props: IPanelStyleProps): IPanelStyles => {
+  const {
+    className,
+    focusTrapZoneClassName,
+    hasCloseButton,
+    headerClassName,
+    isAnimating,
+    isFooterAtBottom,
+    isFooterSticky,
+    isOnRightSide,
+    isOpen,
+    isHiddenOnDismiss,
+    theme,
+    type
+  } = props;
+  const { palette } = theme;
+
+  const classNames = getGlobalClassNames(GlobalClassNames, theme);
+
+  return {
+    root: [
+      classNames.root,
+      isOpen && classNames.isOpen,
+      hasCloseButton && classNames.hasCloseButton,
+      {
+        pointerEvents: 'none',
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0
+      },
+      !isOpen &&
+        !isAnimating &&
+        isHiddenOnDismiss && {
+          visibility: 'hidden'
+        },
+      className
+    ],
+    overlay: [
+      {
+        pointerEvents: 'none',
+        opacity: 1,
+        cursor: 'pointer',
+        transition: `opacity ${AnimationVariables.durationValue3} ${AnimationVariables.easeFunction1}`
+      },
+      isOpen && {
+        cursor: 'pointer',
+        pointerEvents: 'auto'
+      },
+      isOpen && isAnimating && AnimationClassNames.fadeIn200,
+      !isOpen && isAnimating && AnimationClassNames.fadeOut200
+    ],
+    hiddenPanel: [
+      !isOpen &&
+        !isAnimating &&
+        isHiddenOnDismiss && {
+          visibility: 'hidden'
+        }
+    ],
+    main: [
+      classNames.main,
+      {
+        backgroundColor: palette.white,
+        position: 'absolute',
+        right: 0,
+        width: '100%',
+        bottom: 0,
+        top: 0,
+        overflowX: 'hidden',
+        overflowY: 'auto',
+        WebkitOverflowScrolling: 'touch',
+        selectors: {
+          ['@media (min-width: ' + ScreenWidthMinMedium + 'px)']: {
+            borderLeft: `1px solid ${palette.neutralLight}`,
+            borderRight: `1px solid ${palette.neutralLight}`,
+            pointerEvents: 'auto',
+            width: panelSize.width.sm,
+            boxShadow: '-30px, 0px, 30px, -30px, .2',
+            left: 'auto'
+          },
+          '$root &': [
+            isOpen && {
+              pointerEvents: 'auto'
+            },
+            type === PanelType.smallFluid && {
+              width: '100%'
+            },
+            type === PanelType.smallFixedNear && {
+              right: 'auto',
+              left: 0,
+              width: panelSize.width.xs,
+              boxShadow: '30px, 0px, 30px, -30px, .2'
+            },
+            type === PanelType.smallFixedFar && {
+              width: panelSize.width.xs,
+              selectors: {
+                ['@media (min-width: ' + ScreenWidthMinMedium + 'px)']: {
```

Comment:
Media Queries are fine here since its page layout related, right? Not sure how else you'd do it easily. Just thinking about @dzearing's mergeStyle conversion email a few months back re: media queries.